lib/addons/cloudwatch-logs/index.ts

```diff
@@ -0,0 +1,110 @@
+import { PolicyDocument } from 'aws-cdk-lib/aws-iam';
+import { Construct } from "constructs";
+import * as iam from "aws-cdk-lib/aws-iam";
+import { setPath } from "../../utils";
+import { HelmAddOn, HelmAddOnUserProps } from "../helm-addon";
+import { ClusterInfo, Values } from "../../spi/types";
+import { createNamespace } from "../../utils/namespace-utils";
+import { getCloudWatchLogsPolicyDocument } from "./iam-policy";
+
+/**
+ * Configuration options for the FluentBit add-on.
+ */
+export interface CloudWatchLogsAddonProps extends HelmAddOnUserProps {
+    /**
+     * Iam policies for the add-on.
+     */
+    policyDocumentProvider?: (partition: string) => PolicyDocument;
```

Comment:
Why is this here? Can you describe a case when customer will pass some additional policies here?
Also I don't see that you honor any input in this field from the customers. Regardless of what they provide you will apply your own.